tests/utils.tests.js

```diff
@@ -0,0 +1,79 @@
+const axios = require('axios');
+const chai = require('chai');
+const sinon = require('sinon');
+const utils = require('../src/utils');
+
+require('../src/logger');
+
+const expect = chai.expect;
+
+describe('matrixUtils', function() {
+    describe('axiosGet', () => {
+        let axiosStub;
+
+        afterEach(() => {
+            axiosStub.restore();
+        });
+
+        it('ensures domain is not blacklisted', async() => {
+            axiosStub = sinon.stub(axios, 'get');
+            let raised;
+            try {
+                await utils.axiosGet('http://127.0.0.1');
+                raised = false;
+            } catch (error) {
+                raised = true;
+            }
+            expect(raised).to.be.true;
+            expect(axiosStub.called).to.be.false;
+        });
+
```

Comment:
I think a test that the redirection blacklisting is working as expected would be nice. Something like:

```suggestion

it('ensures redirection domain is not blacklisted', async() => {
    axiosStub = sinon.stub(axios, 'get');
                                                                 
    axiosStub.onFirstCall().returns({
        headers: {
            location: 'http://127.0.0.1',
        },
        status: 301,
    });
    axiosStub.onSecondCall().returns({status: 200});
                                                                 
    let raised;
    try {
        await utils.axiosGet('https://matrix.org');
        raised = false;
    } catch (error) {
        raised = true;
    }
                                                                 
    expect(raised).to.be.true;
    expect(axiosStub.calledOnce).to.be.true;
    expect(axiosStub.calledTwice).to.be.false;
});

```